.github/workflows/continuous-integration.yml

```
@@ -34,20 +38,26 @@ jobs:
+        # The default: '[]' for inputs.old_stable applies if it's omitted in workflow_call.
+        # For push/pull_request, inputs.old_stable will be an empty string, so env.DEFAULT_OLD_STABLE_JSON is used.
+        php-version: ${{ fromJson(inputs.old_stable || env.DEFAULT_OLD_STABLE_JSON) }}
         dependencies: [highest, lowest]
         os: [ubuntu-latest]
         experimental: [false]
         include:
-          - php-version: ${{ inputs.current_stable }} # Should be 8.4 for the special handling
+          # If inputs.current_stable is not an empty string (i.e., came from workflow_call or workflow_dispatch), use it.
+          # Otherwise (for push/pull_request), use env.DEFAULT_CURRENT_STABLE.
+          # inputs.current_stable is required for workflow_call, so it will always be present for that trigger.
+          - php-version: ${{ inputs.current_stable || env.DEFAULT_CURRENT_STABLE }}
```

actionlint (1.7.7)

context "env" is not allowed here. available contexts are "github", "inputs", "needs", "vars". see https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability for more details

(expression)